src/server/config.ts

```diff
@@ -68,6 +73,14 @@ export interface DataRequestConfig extends AxiosRequestConfig {
    */
   cacheKey?: string;
   hasBearerToken?: boolean;
+
+  mergeResults?: <T>(
+    responseData: any,
+    newRequest:
+      | ApiSuccessResponse<T>
+      | ApiErrorResponse<null>
+      | ApiPostponeResponse
```

Comment:
Kun je hier ook `ApiResponse` gebruiken?

Reply:
De functie `requestData` voldoet niet aan het type `ApiResponse` omdat de `ApiErrorResponse` in `requestData` altijd gevuld wordt met `null` (ipv `T`). Als ik dat aanpas gaat het op andere plaatsen stuk omdat daar van `null` wordt uitgegaan.